### PR TITLE
Constructable fuse bombs, refactored as siege cannon ammo

### PR DESCRIPTION
**Pull request by ghost**
[tweak][content][tested]
Explosion formula is same as ever, activates when cannonball hits something.
Item entirely refactored as a cannonball instead of a device, moved into the same file too. (there was nothing unique held in device types other than an icon path anyways, which was thankfully set in the item to begin with redundantly)
Instructions on assembling them are in the changelog, let me know if a diamond drill is too much effort to make it with.
:cl:
 * rscadd: Constructable fuse bombs are here! Made with a surgical drill on an iron cannonball, welding fuel and cable coil, these are slightly more destructive than IEDs (without breaching) and can now be used as siege cannon ammo when lit!
 * tweak: Fuse bombs now light their fuse when caught in fires, like they should.

### CODE REVIEW

**Comment by PrimeDSS13** [comment]
maybe it would be better to have this just be a normal drill item
that item gets no use as it is, though I'd like to hear what others think

**Comment by Apogee-dev** [comment]
> maybe it would be better to have this just be a normal drill item
> that item gets no use as it is, though I'd like to hear what others think

normal drill sounds fine to me, gating it behind diamond is a bit lame and given how diamond surgical drills never get made nobody would ever bother with fusebombs if it were left as such

**Comment by ghost** [comment]
> normal drill sounds fine to me, gating it behind diamond is a bit lame and given how diamond surgical drills never get made nobody would ever bother with fusebombs if it were left as such

Replaced with normal drill now

**Comment by Blithering** [comment]
Would these still do the cannonball knockdown/ wall destruction on hit?

**Comment by ghost** [comment]
> Would these still do the cannonball knockdown on hit?

Yes, but for half the length of time, and only if the fuse isn't lit

**Comment by ghost** [comment]
> Would these still do the cannonball wall destruction on hit?

Tested locally, it does

File: code/modules/projectiles/guns/projectile/constructable/siegecannon.dm [hunk 1]
```diff
@@ -278,6 +278,208 @@
...
+
+/obj/item/cannonball/fuse_bomb/proc/fuse_burn()
+	set waitfor = 0
+
+	if(src && src.fuse_lit)
+		if(src.seconds_left)
+			sleep(10)
+			src.seconds_left--
+			src.update_icon()
+			.()
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/projectile/constructable/siegecannon.dm:401]: just one period?

**Pull request merged**